frame/contracts/src/benchmarking/mod.rs

```diff
@@ -1762,6 +1762,60 @@ benchmarks! {
 		let origin = RawOrigin::Signed(instance.caller.clone());
 	}: call(origin, instance.addr, 0u32.into(), Weight::max_value(), vec![])
 
+	// Only calling the function itself with valid arguments.
+	// It generates different private keys and signatures for the message "Hello world"
+	seal_ecdsa_recovery {
+		let r in 0 .. API_BENCHMARK_BATCHES;
+
+		let message_hash = sp_io::hashing::blake2_256("Hello world".as_bytes());
+		let signatures = (0..r * API_BENCHMARK_BATCH_SIZE)
+			.map(|i| {
+				use secp256k1::{SecretKey, Message, sign};
+				use rand::SeedableRng;
+				let mut rng = rand_pcg::Pcg32::seed_from_u64(i as u64);
```

Comment:
I think we can move that out of the closure. No need to reseed every time, right?